src/goto-programs/goto_trace.h

```diff
@@ -138,6 +136,22 @@ class goto_trace_stept
     full_lhs_value.make_nil();
     cond_expr.make_nil();
   }
+
+  goto_programt::const_targett get_pc() const { return pc; }
+
+  /// We don't have a way of enforcing this, but \p pc_ must *not* be a
+  /// singular iterator!
+  void set_pc(goto_programt::const_targett pc_)
+  {
+    pc = std::move(pc_);
+    is_pc_set = true;
+  }
+
+  bool get_is_pc_set() const { return is_pc_set; }
```

Comment:
> So how would this be any worse than the proposed set_pc(...)?

Because it means the class is no-longer default-constructible, which in turn limits how it can be used.

> The use of smart pointers may be an option, if done consistently across the code base for all uses of (const_)targett. But working on that would be an interesting choice of priorities.

Quite, which is why I'm proposing this solution.